mock-thunderhead/src/main/java/com/sequenceiq/thunderhead/grpc/service/auth/MockUserManagementService.java

```diff
@@ -590,7 +595,10 @@ public void getAccount(GetAccountRequest request, StreamObserver<GetAccountRespo
                                 .addEntitlements(createEntitlement(DATAHUB_AZURE_AUTOSCALING))
                                 .addEntitlements(createEntitlement(DATAHUB_AWS_AUTOSCALING))
                                 .addEntitlements(createEntitlement(LOCAL_DEV))
+                                .addEntitlements(createEntitlement(CDP_CP_CUSTOM_DL_TEMPLATE))
                                 .setPasswordPolicy(workloadPasswordPolicy)
+                                .setAccountId(request.getAccountId())
+                                .setExternalAccountId("external-" + request.getAccountId())
```

Comment:
is this ok to using a fixed external name here?

Reply:
Honestly I do not know what this is supposed to be like. For most of the real UMS users, the externalAccountId is derived from the corresponding SalesForce (SFDC) account ID. For instance, mow-dev (accountId=9d74eee4-1cad-45d7-b645-7ccf9edbb73d) has externalAccountId=0018000000ik5epAAA. As far as I can tell, the `001` is a fixed prefix, while the last 3 characters are a sort-of checksum calculated from the rest of the string.

The relevant code in DatalakeApi svc making the check is located [here](https://github.infra.cloudera.com/thunderhead/thunderhead/blob/master/services/api/datalake/src/main/java/com/cloudera/thunderhead/service/datalakeapi/server/DatalakeApiImpl.java#L1102-L1105). The underlying implementation currently only asserts that the account response contains both accountId and externalAccountId and they are non-empty.

I could find a Java implementation for the calculation logic of "real" externalAccountId's based on the SFDC ID in [thunderhead](https://github.infra.cloudera.com/thunderhead/thunderhead/blob/master/services/libs/common-test/src/main/java/com/cloudera/thunderhead/service/common/test/sfdc/SFDCUtils.java#L30-L46), but did not feel it justified to duplicate the same behavior here until it is crucial. What is your preference?